sync-core/src/main/java/com/cloudant/sync/datastore/BasicDatastore.java

```diff
@@ -857,15 +887,7 @@ public Object call(SQLDatabase db) throws Exception {
         } catch (InterruptedException e) {
             throw new RuntimeException(e);
         } catch (ExecutionException e) {
-            if(e.getCause() != null){
-                if(e.getCause() instanceof DocumentNotFoundException){
-                    throw (DocumentNotFoundException)e.getCause();
-                } else {
-                    throw new RuntimeException(e);
-                }
-            } else {
-                throw new RuntimeException(e);
-            }
+           throw new DocumentNotFoundException(String.format("Could not find document with id: %s",docId),e);
```

Comment:
I've always had the impression that getting the cause of the exception and throwing that is bad practise, perhaps if we have :
```
        try {
            int rowsDeleted = queue.submit(new SQLQueueCallable<Integer>() {
                @Override
                public Object call(SQLDatabase db) throws Exception {
                    String[] whereArgs = {docId};
                    return db.delete("localdocs", "docid=? ", whereArgs);
                }
            }).get();
           if (rowsDelete == 0 ) {
                throw new DocumentNotFoundException(String.format("Could not find document with id: %s",docId),e);
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        } catch (ExecutionException e) {
           throw new DocumentException(e)
        }
``` 
it would avoid `DocumentNotFoundException` being the cause of a `DocumentNotFoundException`